File: content/search/source/includes/view/code-snippets/csharp/DecimalToDoubleQuery.cs
using MongoDB.Bson;
using MongoDB.Driver;

// Connect to your Atlas deployment
var uri = "<connectionString>";
var client = new MongoClient(uri);
var database = client.GetDatabase("sample_airbnb");
var collection = database.GetCollection<BsonDocument>(
    "listingsAndReviews");

// Run the aggregation query
var pipeline = new[]
{
    new BsonDocument("$search",
        new BsonDocument("index", "listingsSearchablePrice")
            .Add("range",
                new BsonDocument("path", "totalPrice")
                    .Add("gte", 100)
                    .Add("lte", 200))),
    new BsonDocument("$project",
        new BsonDocument("_id", 0)
            .Add("totalPrice", 1)
            .Add("price", 1)
            .Add("cleaning_fee", 1))
};

var results = collection.Aggregate<BsonDocument>(pipeline);
foreach (var doc in results.ToList())
{
    Console.WriteLine(doc);
}

File: content/search/source/includes/view/code-snippets/python/decimal-to-double-query.py
from pymongo import MongoClient


def query_view():
    # Connect to your Atlas deployment
    uri = "<connectionString>"
    client = MongoClient(uri)

    # Access your database and collection
    database = client["sample_airbnb"]
    collection = database["listingsAndReviews"]

    # Run the aggregation query
    pipeline = [
        {
            "$search": {
                "index": "listingsSearchablePrice",
                "range": {
                    "path": "totalPrice",
                    "gte": 100,
                    "lte": 200
                }
            }
        },
        {
            "$project": {
                "_id": 0,
                "totalPrice": 1,
                "price": 1,
                "cleaning_fee": 1
            }
        }
    ]

    results = collection.aggregate(pipeline)
    for doc in results:
        print(doc)


query_view()
